tensorflow/core/common_runtime/process_util.cc

```diff
@@ -49,8 +51,22 @@ thread::ThreadPool* ComputePool(const SessionOptions& options) {
 int32 NumInterOpThreadsFromSessionOptions(const SessionOptions& options) {
   const int32 t = options.config.inter_op_parallelism_threads();
   if (t != 0) return t;
+#ifdef INTEL_MKL
+  // MKL library executes ops in parallel using OMP threads
+  // Set inter_op conservatively to avoid thread oversubscription that could 
+  // lead to severe perf degradations and OMP resource exhaustion
+  const int mkl_intra_op = omp_get_max_threads();
+  CHECK_GE(mkl_intra_op, 1);
+  const int32 mkl_inter_op = std::ceil(
```

Comment:
I completely agree that we want to avoid thread oversubscription, which can be very severe with the current (silly) default settings, but this seems a bit too pessimistic/conservative. Do you assume that omp_get_max_threads() is significantly smaller than the number of cores on the machine? If not (i.e. if mkl_inter_op ends up being 1), you may hurt performance for models that have a mixture of large expensive and many cheaper ops. If the model is completely dominated by a few expensive, heavily threaded ops, it's fine, of course. There is no simple way of solving this in general without a global scheduling strategy, i.e. the inter and intra threadpools being combined or being aware of each other.

Would you agree that setting inter op equal to

```max(ceil(num_cores / intra), 2)```

would be a reasonable heuristic, or do you have performance data supporting your choice?

Reply:
Agree that this is conservative. For most of the models we have worked with, inter_op 1 seems to be good enough until data layer comes in to the picture at which point, inter_op 2 or higher seems better. I'm willing to bump it up to 2 though.

Reply:
OK, sounds good. Let's bump it to two for now then.